module/zfs/zil.c

```diff
@@ -2595,7 +2613,8 @@ zil_async_to_sync(zilog_t *zilog, uint64_t foid)
 		 */
 		t = &itxg->itxg_itxs->i_async_tree;
 		if (foid != 0) {
-			ian = avl_find(t, &foid, &where);
+			ian_search.ia_foid = foid;
+			ian = avl_find(t, &ian_search, &where);
```

Comment:
It looks like this same issue exists in `zil_remove_async()`.  Let's get both.

Reply:
Done.